File: getDownscaleList.m
function output = getDownscaleList(input, max_downscale)
    
    if strcmp(max_downscale, 'MAX');
        [h, w] = size(input);
        max_downscale = ceil(log2(min(h, w)));
    end
    
    output = zeros(1,max_downscale+1);
    for i = 0: max_downscale
        output(i+1) = 2.^i;
    end

end

File: getDownscaleList.py
import numpy as np
import math

def getDownscaleList(input, max_downscale):
    
    if max_downscale == 'MAX':
        (h, w) = np.shape(input);
        max_downscale = math.ceil(math.log2(max(h, w)));

    output = np.zeros((1,max_downscale));
    for i in range(max_downscale):
        output[0,i] = pow(2,i);
    
    return output;